app/controllers/settings_controller.rb

```diff
@@ -45,8 +45,12 @@ def edit
       end
 
       settings.each do |name, value|
-        # remove blank values in array settings
-        value.delete_if(&:blank?) if value.is_a?(Array)
+        if value.is_a?(Array)
+          # remove blank values in array settings
+          value.delete_if(&:blank?)
+        else
+          value = value.strip
```

Comment:
I could only craft one (e.g. "enter allowed characters here")...
I also don't *think* it should pose a problem, but I am not sure. Plus I am not fond that this is a viable solution to the general problem, e.g. talking about the work package: I could have many other errors in the hostname and different problems in other fields that will still go unnoticed...

But maybe it is better to start with a solution like you are proposing and then further improve from there...